java/src/jmri/jmrix/can/AbstractCanTrafficController.java

```diff
@@ -302,7 +304,11 @@ public void distributeOneReply(CanReply msg, AbstractMRListener mLastSender) {
         // return a notification via the Swing event queue to ensure proper thread
         Runnable r = newRcvNotifier(msg, mLastSender, this);
         try {
-            javax.swing.SwingUtilities.invokeAndWait(r);
+            if (synchronizeRx) {
+                SwingUtilities.invokeAndWait(r);
```

Comment:
Please use the threading tools in jmri.util.ThreadingUtil instead to synchronize on the layout thread instead of the GUI thread.

Reply:
This code has to be the same as in the parent class that we are overriding: https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrix/AbstractMRTrafficController.java#L928
If you think that code is wrong, then we can change them both, but that should be done in a separate PR and that change has to receive a scrutinous testing as it will be a major change in the threading model.
Please let me know if you want to do that.

Reply:
If this method is, and must, be the same as the overridden class, then don't override it. However, if you are overriding it, you are *explicitly* making the code different, so you can deviate from the parent class. See http://jmri.org/help/en/html/doc/Technical/Threads.shtml for how you should be ensuring code runs on the AWT thread.

Reply:
done